marklogic-data-hub-central/src/test/java/com/marklogic/hub/central/entities/search/EntitySearchManagerTest.java

```diff
@@ -274,6 +279,39 @@ void testGetQueryOptions() {
         assertThrows(RuntimeException.class, () -> new EntitySearchManager(getHubClient()).getQueryOptions("non-existent-options"), "Search options doesn't exist");
     }
 
+    @Test
+    void testDownloadRecord() {
```

Comment:
Just test the controller instead via an AbstractMvcTest - that gives us coverage over the controller as well, which is missing right now.

Should test downloading a binary as well - that could be as simple as capturing the bytes of the string "hello world" and storing that as a document. Can assert on its size too.

Reply:
Also, no need to install the reference project and create customer entities. The "download" feature doesn't care about whether the doc is an entity or not - it just takes a URI as an input and returns the document in the HTTP response output stream. 

For testing then, I'd have 5 separate tests - one a JSON doc, one for XML, one for Text, one for Binary, and one for a missing URI (verify you get an appropriate error message).